sklearn/cluster/_k_means.pyx

```diff
@@ -14,6 +14,8 @@ import scipy.sparse as sp
 cimport numpy as np
 cimport cython
 from cython cimport floating
+from cpython cimport array as c_array
```

Comment:
Can you try adding just ``import array`` and see if that works, too? that's how we usually do it, I think.